src/openrct2/world/Footpath.cpp

```diff
@@ -524,55 +524,50 @@ static void footpath_connect_corners(int32_t initialX, int32_t initialY, TileEle
         return;
 
     tileElement[0] = initialTileElement;
-    int32_t z = initialTileElement->base_height;
+    int32_t z = initialTileElement->GetBaseZ();
     for (int32_t initialDirection = 0; initialDirection < 4; initialDirection++)
     {
-        int32_t x = initialX;
-        int32_t y = initialY;
-        int32_t direction = initialDirection;
+        auto currentPos = footpathPos;
 
-        x += CoordsDirectionDelta[direction].x;
-        y += CoordsDirectionDelta[direction].y;
-        tileElement[1] = footpath_connect_corners_get_neighbour(x, y, z, (1 << direction_reverse(direction)));
+        tileElement[1] = footpath_connect_corners_get_neighbour({ currentPos, z }, (1 << direction_reverse(initialDirection)));
         if (tileElement[1] == nullptr)
             continue;
 
-        direction = (direction + 1) & 3;
-        x += CoordsDirectionDelta[direction].x;
-        y += CoordsDirectionDelta[direction].y;
-        tileElement[2] = footpath_connect_corners_get_neighbour(x, y, z, (1 << direction_reverse(direction)));
+        auto directionNext = (initialDirection + 1) & 3;
```

Comment:
I meant use the `direction_next()` function.

Reply:
Argh, sure, my bad,